src/lib/services/segment-service.ts

```diff
@@ -97,11 +109,46 @@ export class SegmentService {
 
     // Used by unleash-enterprise.
     async addToStrategy(id: number, strategyId: string): Promise<void> {
+        await this.validateStrategySegmentLimit(strategyId);
         await this.segmentStore.addToStrategy(id, strategyId);
     }
 
     // Used by unleash-enterprise.
     async removeFromStrategy(id: number, strategyId: string): Promise<void> {
         await this.segmentStore.removeFromStrategy(id, strategyId);
     }
+
+    private async validateStrategySegmentLimit(
+        strategyId: string,
+    ): Promise<void> {
+        const limit = this.experimental?.segments?.strategySegmentsLimit;
+
+        if (typeof limit === 'undefined') {
+            return;
+        }
+
+        if ((await this.getByStrategy(strategyId)).length >= limit) {
+            throw new BadDataError(
+                `Strategies may not have more than ${limit} segments`,
+            );
+        }
+    }
+
+    private validateSegmentValuesLimit(segment: Omit<ISegment, 'id'>): void {
+        const limit = this.experimental?.segments?.segmentValuesLimit;
+
+        if (typeof limit === 'undefined') {
```

Comment:
Just a double check here, is it enough to guard against undefined and not nulls?

Reply:
Yeah, the config field can be undefined or a number, so should be enough.

Reply:
NIce!